src/coreclr/System.Private.CoreLib/src/System/MulticastDelegate.cs

```diff
@@ -428,22 +428,19 @@ public sealed override Delegate[] GetInvocationList()
             return del;
         }
 
-        // Force inline as the true/false ternary takes it above ALWAYS_INLINE size even though the asm ends up smaller
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool operator ==(MulticastDelegate? d1, MulticastDelegate? d2)
         {
             // Test d2 first to allow branch elimination when inlined for null checks (== null)
             // so it can become a simple test
             if (d2 is null)
             {
-                // return true/false not the test result https://github.com/dotnet/runtime/issues/4207
-                return (d1 is null) ? true : false;
+                return d1 is null;
             }
 
             return ReferenceEquals(d2, d1) ? true : d2.Equals((object?)d1);
```

Comment:
```suggestion
            return ReferenceEquals(d2, d1) || d2.Equals((object?)d1);
````

Reply:
This is quite a similar pattern to the one addressed by this PR. It's better to confirm there's no codegen regression separately.